### PR TITLE
Use typed MC Link client

### PR DESCRIPTION
**Pull request by zaharidichev**
This PR switches to using the typed API for accessing the Linkerd Multicluster CRDs.
This PR is blocked on merging: https://github.com/linkerd/linkerd2/pull/7335

Signed-off-by: Zahari Dichev <zaharidichev@gmail.com>

### CODE REVIEW

File: agent/pkg/k8s/helpers_test.go [hunk 1]
```diff
@@ -11,38 +11,34 @@ import (
 	tsfake "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned/fake"
 	tsscheme "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned/scheme"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/dynamic"
```
Comment (siggy) [inline agent/pkg/k8s/helpers_test.go:14]: feels good to ditch `dynamic` 👍

**Review by siggy — APPROVED** [review]
looks good, works well. 👍 🚢

File: go.mod [hunk 2]
```diff
@@ -9,12 +9,12 @@ require (
...
-	google.golang.org/grpc v1.41.0
+	google.golang.org/grpc v1.42.0
 	google.golang.org/grpc/cmd/protoc-gen-go-grpc v1.1.0
 	google.golang.org/protobuf v1.27.1
-	k8s.io/api v0.22.1
-	k8s.io/apimachinery v0.22.1
-	k8s.io/client-go v0.22.1
+	k8s.io/api v0.22.3
+	k8s.io/apimachinery v0.22.3
+	k8s.io/client-go v0.22.3
```
Comment (siggy) [inline go.mod:17]: any reason not to rev `k8s.io/kubectl` to `v0.22.3` as well?

File: go.mod [hunk 3]
```diff
@@ -79,35 +71,30 @@ require (
...
-	k8s.io/kube-aggregator v0.22.1 // indirect
+	helm.sh/helm/v3 v3.7.1 // indirect
+	k8s.io/apiextensions-apiserver v0.22.3 // indirect
+	k8s.io/kube-aggregator v0.22.3 // indirect
 	k8s.io/kube-openapi v0.0.0-20210817084001-7fbd8d59e5b8 // indirect
 	k8s.io/utils v0.0.0-20210820185131-d34e5cb4466e // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
 )
+
+replace github.com/linkerd/linkerd2 v0.5.1-0.20211101210204-ff8209b9cc86 => github.com/zaharidichev/linkerd2 v0.5.1-0.20211122083058-d0fccb73fb7a
```
[the remark this answers is not in the document]
Comment (zaharidichev) [reply go.mod:100]: Yes, correct

**Pull request merged**